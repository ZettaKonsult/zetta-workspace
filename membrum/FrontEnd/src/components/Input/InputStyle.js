import styled from 'styled-components'

const InputStyle = styled.input`
  padding: 15px;
`

export default InputStyle
