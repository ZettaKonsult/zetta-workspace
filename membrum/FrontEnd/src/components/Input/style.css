.InputField {
  display: flex;
  align-items: center;
  max-width: 1200px;
}

.InputField > label {
  flex: 0 1 200px;
  font-weight: 300;
  letter-spacing: 0.09em;
  text-transform: uppercase;
  text-align: left;
}

.InputField > input {
  flex: 1 0 auto;
  padding: 15px;
  align-self: stretch;
}

@media (max-width: 500px) {
  .InputField {
    flex-flow: column nowrap;
    align-items: flex-start;
  }

  .InputField > label {
    flex: 1 0 auto;
    margin-bottom: 0.2em;
  }
}
