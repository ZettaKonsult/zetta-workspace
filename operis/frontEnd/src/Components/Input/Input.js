import 'react'
import styled from 'styled-components'

const Input = styled.input`
  padding: 0.5em;
  font-size: 1.3em;
  width: 90%;
  overflow: visible;
`

export default Input
