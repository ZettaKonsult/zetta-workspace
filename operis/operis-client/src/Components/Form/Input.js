import React from 'react';

import { Form } from 'semantic-ui-react';

export const Input = ({
  input,
  type,
  name,
  placeholder,
  meta: { touched, error, warning },
}) => (
  <div>
    <Form.Field error={!!touched && !!error}>
      <label>{placeholder}</label>
      <input {...input} type={type} placeholder={placeholder} />
      {!!touched &&
        !!error && <span style={{ color: '#ae5856' }}>{error}</span>}
    </Form.Field>
  </div>
);
