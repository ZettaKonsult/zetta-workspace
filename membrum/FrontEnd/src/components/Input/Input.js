import React from 'react'
import PropTypes from 'prop-types'
import InputStyle from './InputStyle'
const Input = ({ name, value, onChange, id, type, label }) => (
  <InputStyle
    type={type}
    name={name}
    placeholder={name}
    value={value}
    onChange={onChange}
  />
)

Input.propTypes = {
  name: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func.isRequired,
  type: PropTypes.string
}

Input.defaultProps = {
  name: 'default',
  type: 'text'
}

export default Input
