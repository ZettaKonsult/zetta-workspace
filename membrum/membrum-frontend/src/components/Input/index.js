import React from "react"
import PropTypes from "prop-types"
import "./style.css"

const Input = ({ name, value, onChange, id, type, label }) => (
  <div className="InputField">
    <label htmlFor={name}>{label ? label : name}:</label>
    <input
      type={type}
      id={id ? id : name}
      name={name}
      placeholder={name}
      value={value}
      onChange={onChange}
    />
  </div>
)

Input.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  type: PropTypes.string
}

Input.defaultProps = {
  name: "default",
  type: "text"
}

export default Input
