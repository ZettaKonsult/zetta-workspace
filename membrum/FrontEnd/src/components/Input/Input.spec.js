import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';

it('should render a Input', function() {
  const tree = renderer.create(<Input onChange={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
