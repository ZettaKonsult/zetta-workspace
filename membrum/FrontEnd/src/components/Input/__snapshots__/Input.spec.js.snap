// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Input 1`] = `
<input
  className="sc-bdVaJa dgUUwq"
  name="default"
  onChange={[Function]}
  placeholder="default"
  type="text"
  value={undefined}
/>
`;
